library/src/lib/alert/alert.component.ts

```diff
@@ -1,137 +1,175 @@
+            } else if (this.childComponentType instanceof TemplateRef) {
+                this.loadFromTemplate(this.childComponentType);
+            } else {
+                this.loadFromString(this.childComponentType);
+            }
+            this.cdRef.detectChanges();
         }
     }
 
-    handleClose(result?, fromService?, timeout?) {
-        if (!timeout && timeout !== 0) {
-            timeout = 750;
+    dismiss(manualDismiss: boolean = false): void {
+        if (manualDismiss) {
+            this.elRef.nativeElement.style.display = 'none';
         }
-        this.show = false;
-        this.close.emit(this.id);
-        setTimeout(() => {
+        if (this.alertRef) {
+            this.alertRef.dismiss();
+        } else {
             this.elRef.nativeElement.style.display = 'none';
-            this.alertDiv.nativeElement.style.display = 'none';
-            if (!fromService && !this.inline) {
-                this.alertService.popAlert();
-            }
-            this.afterClosed.next();
-        }, timeout)
+        }
+        this.close.emit();
     }
 
-    open() {
-        // check to make sure this alert is not already opened
-        if (this.elRef.nativeElement.style.display !== 'block') {
-            this.show = true;
-            this.cd.detectChanges();
-            const top = this.getTop();
-            this.elRef.nativeElement.style.display = 'block';
-            this.alertDiv.nativeElement.style.display = 'block';
-            this.alertDiv.nativeElement.style.top = top;
-            if (!this.persist) {
-                setTimeout(() => {
-                    if (this.mousePersist) {
-                        const wait = () => {
-                            if (this.mouseInAlert === true) {
-                                setTimeout(wait, 500);
-                            } else {
-                                this.handleClose();
-                            }
-                        }
-                        wait();
-                    } else {
-                        this.handleClose();
-                    }
-                }, this.visibleTime);
+    open(): void {
+        if (!this.alertRef) {
+            if (this.elRef.nativeElement.style.display === 'block') {
+                return;
             }
+            this.elRef.nativeElement.style.display = 'block';
         }
-    }
 
-    getTop() {
-        // get the heights of each visible alert and return the top of the new alert
-        const alerts = document.querySelectorAll('.fd-alert');
-        const openAlerts = Array.prototype.filter.call(alerts, function(alert) {
-            return alert.style.display === 'block';
-        });
-        let totalOffsetHeight = 10;
-        openAlerts.forEach((alert) => {
-            totalOffsetHeight = totalOffsetHeight + alert.offsetHeight + 10;
-        });
-        return totalOffsetHeight + 'px';
+        if (!this.persist) {
+            setTimeout(() => {
```

Comment:
this timeout should short-circuit when someone calls dismiss on the alertRef

Reply:
Can you please clarify?

Reply:
I thought/think maybe this was the cause of the twice-called destraoyContainer.
If you keep a reference to the timeout `const timeoutKey = window.setTimeout....` you can cancel it with `window.clearTimeout(timeoutKey)`
And just intercept someone calling dismiss on the alertRef, and stop this timeout right away

Reply:
Well it **is** the cause of the twice-called destroyContainer, but I don't really want to manage a setTimeout reference. I would need to keep it inside a component attribute, and that just seems like a bad practice. Like I mentioned below, I added a check for the container and it's sufficient for this. There could be a better solution though.